### ISSUE TITLE
Flag echo usage errors when selected(), disabled(), checked() helpers

### ISSUE THREAD

**Issue by westonruter** [WordPress-Coding-Standards/WordPress-Coding-Standards#346]
Consider:

```php
$output .= selected( in_array( $category->slug, $selected_options ) );
```

This will not have the desired effect, since the 3rd argument to `selected()`, `$echo`, defaults to `true`. 

It would be helpful if there was a sniff that looked at any instance of `selected()`, `disabled()`, `checked()`, etc that have an lvalue (assigned to a variable) if they do not supply the 3rd argument to the function, as such instances are surely accidental omissions of that argument.

The above PHP code could be automatically fixed up to be:

```php
$output .= selected( in_array( $category->slug, $selected_options ), true, false );
```

**Comment by JDGrimes** [comment]
It could also check if `$echo` is set to `false` when inside an `echo` statement (I've been bitten by that before).

**Comment by jrfnl** [comment]
Loosely related: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1341#issuecomment-387196806

**Comment by jrfnl** [comment]
WP 4.9 introduces `readonly()` which works in the same way and should be included in this check.
